tests/integration/deps/test_podman_compose_deps.py

```diff
@@ -183,3 +185,74 @@ def test_deps_fails(self):
                 compose_yaml_path(suffix),
                 "down",
             ])
+
+
+class TestComposeConditionalDepsHealthy(unittest.TestCase, RunSubprocessMixin):
+    def test_up_deps_healthy(self):
+        suffix = "-conditional-healthy"
+        try:
+            self.run_subprocess_assert_returncode([
+                podman_compose_path(),
+                "-f",
+                compose_yaml_path(suffix),
+                "up",
+                "sleep",
+                "--detach",
+            ])
+
+            # The `podman wait --condition=healthy` is invalid prior to 4.6.0.
+            # Since the podman-compose project uses podman 4.3.1 in github actions, we
+            # use sleep as workaround to wait until the `sleep` container becomes running.
+            time.sleep(3)
```

Comment:
I think better idea would be to use polling. Right now all tests have been slowed down by 3 seconds. Additionally, it makes sense to check podman version and do proper test if the version is >= 4.6.0.

Reply:
Agree, this testcase has been branched by podman version.